dips/dip-8.md

```diff
@@ -1,55 +1,144 @@
 ---
 dip: 8
-title: Off-chain API Version 1 Extensions
-authors: Kevin Hurley (@kphfb)
+title: Off-chain API Version 2 Extensions - Funds pull pre approval
+authors: Kevin Hurley (@kphfb), Daniel Prinz
 status: Draft
-type: Informational
+type: Standard
 created: 09/18/2020
 ---
 
 # Summary
 ---
-An extension of the Off-Chain protocol to provide support for more advanced merchant use-cases.
+An extension of the Off-Chain protocol to provide support for enabling the notion of periodic resettable limits within a single approval
 
 ---
 # Abstract / Motivation
 ---
 
-Version 0 of the Off-Chain Protocol is described in [DIP 1](https://dip.diem.com/dip-1/).  Version 1 as described here is an extension of the Off-Chain Protocol and adds features to support more advanced functionality - particularly targeted to support merchant use-cases.  This is inclusive of pull payments, recurring payments, and auth/capture flows.
+Versions 1/2 of the Off-Chain Protocol are described in [DIP 1](https://dip.diem.com/dip-1/). 
+
+Version 3 as described here is an extension of the Off-Chain Protocol and adds features to support more advanced functionality - particularly targeted to support merchant use-cases. 
+
+This DIP aimed to support a scenario of setting a Diem wallet as a long term payment method in an app or an eCommerce website. 
+
+The high level flow is:
+1. The user asks to add their wallet to the app as a payment method
+2. The merchant VASP genereates a Funds Pull Pre Approval (FPPA) request 
+3. The user approves the FPA in their wallet (VASP)
+4. The user wallet VASP hands a billing agreement id (FPPA id) to the merchant VASP
+
+Whenever the app wants to charge the user: 
+1. The merchant VASP sends a payment request to the user wallet VASP mentioning the FPA id
+2. If the payment request does not fall out of the FPA scope, the payment is continued normally as the user was approved it (without user interaction)
+
+---
+# Disclaimer
+---
+
+This DIP does not contain the phase of exchanging identifiers between VASPs that is required to negotiate merchant scenarios.
+
+For the purpose of this document, we will assume that given a payment object with two VASP addresses and some underline indetifier, there is a protocol for exchaning these identities under VASPs (buyer and merchant) without revealing their identities to the public. 
+
+The process described below starts at the phase that both sides have the relevant payment unique identifier (reference) and each VASP knows to corrlate it to its respectful entity under. 
+
+---
+# Synonyms
+---
+
+For the sake of simplicity, we use the following terms:
+* "Buyer VASP" - The VASP who submits payment transactions to the blockchain in order to pay to the other VASP
+* "Merchant" - The VASP who recieves the funds (transaction)
```

Comment:
Can we explicitly say Merchant VASP as we have other docs that have a lot of communication between Merchant and Merchant VASP.